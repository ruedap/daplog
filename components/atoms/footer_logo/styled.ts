import styled from 'styled-components'
import Styles from '@/styles'

export const Img = styled.img`
  opacity: ${Styles.funcs.fibo('md', 'alpha')};
  transition: opacity 0.3s;
  &:hover {
    opacity: ${Styles.funcs.fibo('lg', 'alpha')};
  }
`
