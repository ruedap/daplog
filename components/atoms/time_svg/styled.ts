import styled from 'styled-components'
import Styles from '@/styles'

export const Svg = styled.svg`
  display: block;
  left: 0;
  position: absolute;
  top: 0;
`

export const Circle = styled.circle`
  fill: ${({ theme }) => theme.colors.key[3]};
`

export const Text = styled.text`
  ${Styles.mixins.fontSmoothing()};
  fill: #fff;
  font-family: var(--b-fontFamily-fjalla);
  font-size: ${Styles.funcs.fibo('xs', 'rem')};
  letter-spacing: 0.1em;
`
