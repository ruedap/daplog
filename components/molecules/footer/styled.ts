import styled, { css } from 'styled-components'
import Styles from '@/styles'

export const Root = styled.footer`
  display: block;
  position: relative;
`

export const Container = styled.div`
  ${({ theme }) => css`
    ${Styles.mixins.container};
    background-image: linear-gradient(
      ${theme.colors.key[3]} 0,
      ${theme.colors.key[4]} 100%
    );

    ${theme.mq.up.lg} {
      &::before {
        ${Styles.mixins.dapicons};
        color: ${theme.colors.key[2]};
        content: "ruedap";
        font-size: ${theme.fontSizes.md.rem};
        left: 50%;
        position: absolute;
        top: 377px;
        transform: translate(-50%, -100%);
      }
    }
  `}
`

export const Heading = styled.div`
  margin: 0;
  position: relative;
`

export const LogoSpacer = styled.div`
  display: block;
  font-size: 0;
  /* 1129 / 987 = 1.143870314 */
  padding-bottom: 114.387%;
  position: relative;
  width: 100%;
`

export const LogoLink = styled.a`
  bottom: 0;
  display: block;
  height: auto;
  left: 0;
  margin: auto;
  position: absolute;
  right: 0;
  top: 0;
`
