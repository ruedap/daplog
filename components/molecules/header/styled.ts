import styled from 'styled-components'
import Styles from '@/styles'

export const Inner = styled.div`
  ${Styles.mixins.container}
  background-color: #fff;
`

export const Heading = styled.div`
  margin: 0;
  padding-bottom: 11.854%;
`

export const LogoLink = styled.a`
  display: block;
  position: relative;
  width: 100%;
`

export const LogoSpacer = styled.div`
  display: block;
  font-size: 0;
  margin-top: -50%;
  padding-bottom: 100%;
  position: relative;
  width: 100%;
`

export const LogoOuter = styled.div`
  background-color: var(--b-bgColor-body);
  background-image: radial-gradient(var(--b-bgColor-body), var(--b-color-key3));
  border-radius: 100%;
  bottom: 0;
  display: block;
  height: auto;
  left: 0;
  margin: auto;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
`
