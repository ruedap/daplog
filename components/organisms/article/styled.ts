import styled, { css } from 'styled-components'
import Styles from '@/styles'
import * as StyledMarkdown from './_styled_markdown'

export const TimeContainer = styled.div`
  display: block;
  left: 0;
  margin: auto;
  max-width: var(--b-maxWidth-container);
  pointer-events: none;
  position: absolute;
  right: 0;
  top: 0;
  width: 100%;
`

export const TimeSpacer = styled.div`
  display: block;
  font-size: 0;
  padding-bottom: 61.8%;
  position: relative;
  width: 100%;
`

export const Title = styled.h1`
  ${({ theme }) => css`
    ${Styles.mixins.fontSmoothing()}
    font-family: var(--b-fontFamily-crimson);
    font-size: ${Styles.funcs.fibo('xs', 'rem')};
    font-weight: normal;
    line-height: 1.618;
    margin-top: 0;
    margin-bottom: ${Styles.funcs.fibo('sm', 'px')};
    margin-left: auto;
    margin-right: auto;
    max-width: 877px;
    text-align: center;
    width: 90%;

    ${theme.mq.up.md} {
      font-size: ${Styles.funcs.fibo('sm', 'rem')};
      margin-bottom: ${Styles.funcs.fibo('md', 'px')};

      span::after {
        content: "\\000a";
        white-space: pre-line;
      }
    }
  `}
`

export const GitHubLink = styled.a`
  ${({ theme }) => css`
    ${Styles.mixins.articleSectionMark('xs')}
    height: ${Styles.funcs.fibo('xs', 'px')};
    margin-bottom: calc(${Styles.funcs.fibo('sm', 'px')} + 10px);
    width: calc(${Styles.funcs.fibo('xs', 'px')} * 2);

    ${theme.mq.up.md} {
      ${Styles.mixins.articleSectionMark('sm')}
      height: ${Styles.funcs.fibo('sm', 'px')};
      margin-bottom: calc(${Styles.funcs.fibo('md', 'px')} + 10px);
      width: calc(${Styles.funcs.fibo('sm', 'px')} * 2);
    }
  `}
`

export const Body = styled.section`
  margin: 0 auto;
  max-width: 699px;
  width: 90%;
  ${StyledMarkdown.base}
  ${StyledMarkdown.heading}
  ${StyledMarkdown.p}
  ${StyledMarkdown.img}
  ${StyledMarkdown.strong}
  ${StyledMarkdown.list}
  ${StyledMarkdown.hr}
  ${StyledMarkdown.figure}
  ${StyledMarkdown.cite}
  ${StyledMarkdown.blockquote}
  ${StyledMarkdown.small}
  ${StyledMarkdown.footnotes}
  ${StyledMarkdown.highlight}
  ${StyledMarkdown.code}
  ${StyledMarkdown.table}
  /* extra styles */
  ${StyledMarkdown.bookCover}
  ${StyledMarkdown.embed}
  ${StyledMarkdown.dapicons}
`

export const Nav = styled.nav`
  ${({ theme }) => css`
    margin-top: ${Styles.funcs.fibo('xl', 'px')};
    padding-bottom: ${Styles.funcs.fibo('xl', 'px')};

    ${theme.mq.up.md} {
      margin-top: ${Styles.funcs.fibo('xl2', 'px')};
      padding-bottom: ${Styles.funcs.fibo('xl2', 'px')};
    }
  `}
`

export const NavLink = styled.a`
  ${({ theme }) => css`
    ${Styles.mixins.linkColors('#fff', '#fff', theme.colors.key[4], theme.colors.key[4], theme.colors.key[5])}
    ${Styles.mixins.fontSmoothing()}
    ${Styles.mixins.dapicons}
    background-color: ${theme.colors.key[2]};
    border-radius: 100%;
    display: block;
    font-size: 2.6rem;
    width: ${Styles.funcs.fibo('md', 'px')};
    height: ${Styles.funcs.fibo('md', 'px')};
    line-height: ${Styles.funcs.fibo('md', 'px')};
    margin: auto;
    text-align: center;
    text-decoration: none;
  `}
`
