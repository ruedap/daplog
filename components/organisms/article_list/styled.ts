import styled, { css } from 'styled-components'
import Styles from '@/styles'

export const Root = styled.ul`
  margin-top: 0;
  margin-bottom: 0;
  padding-left: 0;
  list-style: none;
  color: #fff;
`

export const Item = styled.li`
  ${Styles.mixins.fontSmoothing()}
  font-size: 1.4rem;
  line-height: ${Styles.funcs.fibo('sm', 'px')};
  margin-bottom: 1px;

  ${Styles.mq.up.md(css`
    font-size: 1.8rem;
    line-height: ${Styles.funcs.fibo('md', 'px')};
  `)}
`

export const YearHeading = styled(Item)`
  ${({ theme }) => css`
    background-color: ${theme.colors.key[3]};
    color: ${theme.colors.key[5]};
    font-family: var(--b-fontFamily-fjalla);
    height: ${Styles.funcs.fibo('sm', 'px')};
    letter-spacing: 0.1em;
    text-align: center;

    ${Styles.mq.up.md(css`
      height: ${Styles.funcs.fibo('md', 'px')};
    `)}
  `}
`

export const Year = styled.span`
  display: none;
`
export const Dot = Year
