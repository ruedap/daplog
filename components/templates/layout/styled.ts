import styled from 'styled-components'
import Styles from '@/styles'

export const Container = styled.div`
  ${Styles.mixins.container}
  background-color: ${({ theme }) => theme.colors.bg.content};
`
