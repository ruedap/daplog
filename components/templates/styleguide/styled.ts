export * from './_styled_tag4'

import styled, { css, keyframes, Keyframes } from 'styled-components'

export const Title = styled.h1`
  font-size: 4rem;
  margin: 0;
  color: ${({ theme }) => theme.colors.primary};
`

const base = css`
  background-color: white;
  color: cornflowerblue;
  border: 1px solid lightgreen;
  border-right: none;
  border-bottom: none;
  box-shadow: 5px 5px 0 0 lightgreen, 10px 10px 0 0 lightyellow;
  transition: all 0.1s linear;
  margin: 0 3rem 3rem;
  padding: 1rem 0.5rem;
`

const hover = css`
  &:hover {
    color: ${({theme}) => theme.colors.primary};
    border-color: aqua;
    box-shadow: -15px -15px 0 0 aqua, -30px -30px 0 0 cornflowerblue;
  }
`

export const prop = css`
  font-size: 200%;
  color: indigo;
`

export const bounce = keyframes`
  from {
    transform: scale(1.01);
  }
  to {
    transform: scale(0.99);
  }
`

export const shake = keyframes`
  10%, 90% { transform: translateY(-1px); }
  20%, 80% { transform: translateY(2px); }
  30%, 50%, 70% { transform: translateY(-4px); }
  40%, 60% { transform: translateY(4px); }
`

export const Tag1 = styled.div`
  ${base};
`

export const Tag2 = styled.div`
  ${base};
  ${hover};
  & code {
    background-color: linen;
  }
`

export const Tag3 = styled.div<{ animation: Keyframes }>`
  ${base};
  ${hover};
  & code {
    background-color: linen;
  }
  animation: ${({ animation }) => animation } 0.2s infinite ease-in-out alternate;
`
