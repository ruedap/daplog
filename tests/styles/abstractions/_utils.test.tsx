import { utils } from '@/styles/abstracts/_utils'
import styled from 'styled-components'
import { renderWithTheme } from '@/tests/helpers'

describe('utils.units()', () => {
  it('should returns converted each value', () => {
    const actual = utils.units(16)
    expect(actual()).toEqual(16)
    expect(actual.px()).toEqual('16px')
    expect(actual.rem()).toEqual('1rem')
  })

  it('should returns converted each value', () => {
    const actual = utils.units(-100)
    expect(actual()).toEqual(-100)
    expect(actual.px()).toEqual('-100px')
    expect(actual.rem()).toEqual('-6.25rem')
  })

  it('should be match snapshot', () => {
    const u = utils.units(10)
    const Styled = styled.div`
      line-height: ${u};
      font-size: ${u.px};
      width: ${u.rem};
    `
    const actual = renderWithTheme(<Styled />).toJSON()
    expect(actual).toMatchSnapshot()
  })
})

describe('utils.pxToRem()', () => {
  it('should returns converted value', () => {
    const actual = utils.pxToRem(20)
    expect(actual).toEqual('1.25rem')
  })
})
