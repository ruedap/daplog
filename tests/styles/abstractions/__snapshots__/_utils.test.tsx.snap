// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils.units() should be match snapshot 1`] = `
.c0 {
  line-height: 10;
  font-size: 10px;
  width: 0.625rem;
}

<div
  className="c0"
/>
`;
